### PR TITLE
Fixed server socket connection not checked properly

### PR DESCRIPTION
**Pull request by mrsombre**
Socket connection is not checked properly. This caused to infinite loop in some situations.

Processes inside docker container:
```
PID TTY      STAT   TIME COMMAND
...
    9 pts/0    S+     0:00 sh -c yarn dredd ...
...
   53 pts/0    S+     0:00 sh vendor/bin/dredd-hooks-php hooks/init.php
   55 pts/0    S+     0:00 php vendor/ddelnano/dredd-hooks-php/bin/dredd-hooks-php hooks/init.php
```

As you can see, original php server process is a spawn child of sh parent inside docker. When dredd send  SIGTERM to php hooks server (which inside docker he think is PID 53) only sh parent is killed, but child keeps alive.

```
info: Hooks handler stderr: PHP Warning:  stream_socket_accept(): accept failed: Connection timed out in vendor/ddelnano/dredd-hooks-php/src/Server.php on line 40

info: Hooks handler stderr: PHP Warning:  stream_socket_recvfrom() expects parameter 1 to be resource, boolean given in vendor/ddelnano/dredd-hooks-php/src/Server.php on line 42
```

After that listener stay in infinite loop.

### CODE REVIEW

File: src/Server.php [hunk 1]
```diff
@@ -22,24 +24,21 @@ public function __construct($host, $port)
...
 
         if ($server === false) {
-
             throw new UnexpectedValueException("Server could not bind to socket: $errorMessage");
         }
 
         $buffer = "";
 
-        for (;;) {
+        while ($connection = @stream_socket_accept($server)) {
```
Comment (ddelnano) [inline src/Server.php]: Remove the `@` this means the user won't see any output when this fails and those warnings are helpful.

**Comment by ddelnano** [comment]
One minor comment, but thanks for fixing this!

**Comment by mrsombre** [comment]
Removed error suppression.
Can you push new tag to packagist, because we try to build API CI on top of dredd and php hooks. Will be great ) Thanks for the helpful lib.

**Comment by ddelnano** [comment]
Yea, will do.

**Pull request merged**